frontend/src/reducer/notificationPage.js

```diff
@@ -286,14 +286,40 @@ export default function notificationPage (state = defaultNotificationsObject, ac
     }
 
     case `${READ}/${NOTIFICATION}`: {
-      const notification = state.list.find(notification => notification.id === action.notificationId && !notification.read)
-      if (!notification) return state
-      const newUnreadMentionCount = notification.type === `${TLM_ET.MENTION}.${TLM_CET.CREATED}` ? state.unreadMentionCount - 1 : state.unreadMentionCount
+      let notification
+      let replaceList
+      let newUnreadMentionCount
+
+      // NOTE - MP - 07-03-2022 - This is code allow me to read a notification even in a group
+      // This will be fix with the notification refactorisation #5497
+      for (const element of state.list) {
+        if (element.group) {
+          notification = element.group.find(n => n.id === action.notificationId && !n.read)
+          if (notification) {
+            newUnreadMentionCount = (notification.type === `${TLM_ET.MENTION}.${TLM_CET.CREATED}`) ? state.unreadMentionCount - 1 : state.unreadMentionCount
+            element.group = element.group.map(n => n.id === action.notificationId ? { ...notification, read: true } : n)
+            replaceList = state.list.map(g => g.group && g.id === element.id ? element : g)
+            break
+          }
+        }
+      }
```

Comment:
this comment isn't required change since the code is subject to be refactored.
Avoid naming your iterator "element", it's too generic. If you can't name it, it's usually a sign a bad architecture (which makes since since a refactor is planed).
Here, I'm assuming you can't name it because it can be 2 things: notification or group.
My solution would be explicit and name your iterator `notificationOrGroup`.
Also, avoid for loop with break. Since you only want to work with state.list that has a group property, you can do:
``` js
const notif = state.list
  .filter(notifOrGroup => notifOrGroup.group)
  .find(group => group.group.find(n => n.id === action.notificationId && !n.read)
if (notif) {
  newUnreadMentionCount = (notification.type === `${TLM_ET.MENTION}.${TLM_CET.CREATED}`) ? state.unreadMentionCount - 1 : state.unreadMentionCount
  element.group = element.group.map(n => n.id === action.notificationId ? { ...notification, read: true } : n)
  replaceList = state.list.map(g => g.group && g.id === element.id ? element : g)
}
```

Reply:
To clarify, I'm not telling you to never use for loops, I'm just not a fan of for loops that has a break.
And I like pushing my team to use immutable array manipulation functions (.map(), .some(), .filter() ...) instead of for loops.